tests/aws/scenario/test_loan_broker.py

```diff
@@ -0,0 +1,259 @@
+import aws_cdk as cdk
+import aws_cdk.aws_dynamodb as dynamodb
+import aws_cdk.aws_lambda as awslambda
+import aws_cdk.aws_logs as logs
+import aws_cdk.aws_stepfunctions as sfn
+import aws_cdk.aws_stepfunctions_tasks as tasks
+import pytest
+
+from localstack.testing.aws.util import is_aws_cloud
+from localstack.testing.pytest import markers
+from localstack.testing.scenario.provisioning import InfraProvisioner
+from localstack.utils.strings import short_uid
+from localstack.utils.sync import retry
+
+RECIPIENT_LIST_STACK_NAME = "LoanBroker-RecipientList"
+PROJECT_NAME = "CDK Loan Broker"
+OUTPUT_LOAN_BROKER_STATE_MACHINE_ARN = "LoanBrokerArn"
+OUTPUT_LOAN_BROKER_LOG_GROUP_NAME = "LogGroupName"
+LOAN_BROKER_TABLE = "LoanBrokerBanksTable"
+
+CREDIT_BUREAU_JS = "./resources_loan_broker/bank_app_credit_bureau.js"
+BANK_APP_JS = "./resources_loan_broker/bank_app.js"
+
+
+def _read_file_as_string(filename: str):
+    file_path = os.path.join(os.path.dirname(__file__), filename)
+
+    content = None
+    with open(file_path, "r") as file:
+        content = file.read()
+    return content
+
+
+@dataclass
+class Bank:
+    bank_id: str
+    base_rate: str
+    max_loan: str
+    min_credit_score: str
+
+    def get_env(self) -> str:
+        return {
+            "BANK_ID": self.bank_id,
+            "BASE_RATE": self.base_rate,
+            "MAX_LOAN_AMOUNT": self.max_loan,
+            "MIN_CREDIT_SCORE": self.min_credit_score,
+        }
+
+
+class TestLoanBrokerScenario:
+    BANKS = {
+        "BankRecipientPawnShop": Bank(
+            bank_id="PawnShop", base_rate="5", max_loan="500000", min_credit_score="400"
+        ),
+        "BankRecipientUniversal": Bank(
+            bank_id="Universal", base_rate="4", max_loan="700000", min_credit_score="500"
+        ),
+        "BankRecipientPremium": Bank(
+            bank_id="Premium", base_rate="3", max_loan="900000", min_credit_score="600"
+        ),
+    }
+
+    @pytest.fixture(scope="class", autouse=True)
+    def infrastructure(self, aws_client):
+        infra = InfraProvisioner(aws_client)
+        app = cdk.App()
+        recipient_stack = cdk.Stack(app, RECIPIENT_LIST_STACK_NAME)
+        cdk.Tags.of(recipient_stack).add("Project", PROJECT_NAME)
+        cdk.Tags.of(recipient_stack).add("Stackname", RECIPIENT_LIST_STACK_NAME)
+        self.setup_recipient_list_stack(recipient_stack)
+
+        infra.add_cdk_stack(recipient_stack)
+
+        # set skip_teardown=True to prevent the stack to be deleted
+        with infra.provisioner(skip_teardown=True) as prov:
+            if not infra.skipped_provisioning:
+                # here we could add some initial setup, e.g. pre-filling the app with data
+                bank_addresses = [{"S": bank_name} for bank_name in self.BANKS.keys()]
+                aws_client.dynamodb.put_item(
+                    TableName=LOAN_BROKER_TABLE,
+                    Item={"Type": {"S": "Home"}, "BankAddress": {"L": bank_addresses}},
+                )
```

Comment:
An alternative here would be to do this in a test itself, since you could also consider these operations a sort of scenario on the infrastructure :thinking: 
Let's discuss this offline later :+1:

Reply:
@dominikschubert: moved this part, and included some additional testing with dynamodb (update/get-item/scan/delete-item)